packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -219,6 +229,7 @@ export function dispatchEvent(
         nativeEvent,
       )
     ) {
+      nativeEvent.stopPropagation();
```

Comment:
I think we can get into this branch when `enableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay` is `false` but `isDiscreteEventThatRequiresHydration()` also returns `false`. So this potentially changes the behavior outside the feature flag. Is that intentional?